ote/src/clj/ote/services/transit_changes.clj

```diff
@@ -160,6 +164,12 @@
           (gtfs-tasks/detect-new-changes-task db (time/now) true)
           "OK"))
 
+  ;; Force detection change for all services
+  (GET "/transit-changes/detect-changes-for-given-service/:service-id" [service-id :as {user :user}]
```

Comment:
1. This basically same as following existing endpoint:
```
;; Calculate date-hashes for given service-id and package-count
  (GET "/transit-changes/force-calculate-hashes/:service-id/:package-count" [service-id package-count :as {user :user}]
```

Would it bring benefit to try to follow RESTful principles here as well?
In which case this would be nd "overload" to force-calculate-hashes, something like:
``` 
 ;; Calculate date-hashes for given service-id and package-count
  (GET "/transit-changes/force-calculate-hashes/:service-id" [service-id :as {user :user}]
    (when (authorization/admin? user)
      (do
        ;; Calculate date hashes in other thread
        (future
          (detection/calculate-package-hashes-for-service db (Long/parseLong service-id) (Long/parseLong package-count) (:user user))
          ;; Detect changes for service
          (gtfs-tasks/detect-new-changes-task db (time/now) true [(Long/parseLong service-id)]))
"OK")))
```

2. Note that `force-calculate-hashes` uses `future` threading to perform the same task while this doesn't. Probably this should - which gets us to comment part 1. here :)

Reply:
Okay well this initial comment was written after only a quick look so the proposed implementation is not fully kosher and well thought, but pleaase consider the API point of view :)

Reply:
Force-calculate-hashes is different than detect-changes.
So it is better to refactor those API endpoints to make this more clear.